src/main/java/com/learning/pages/PageBase.java

```diff
@@ -0,0 +1,12 @@
+package com.learning.pages;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.PageFactory;
+
+public abstract class PageBase {
+  public PageBase(WebDriver driver) {
+    PageFactory.initElements(driver, this);
+  }
```

Comment:
@PostConstruct is deprecated since Java 11.
What do you think about this solution?

Reply:
It looks fine, I'm however not sure if there won't be any side effects.

Reply:
it seems to not be deprecated, it was removed from sdk -> https://stackoverflow.com/a/55622713/2838206

But... it also looks like Guice does not support those annotations... so you'd have to figure out a different way to manage object lifecycle.

Reply:
> It looks fine, I'm however not sure if there won't be any side effects.

What do you mean? What are possible side effects?

Reply:
> it seems to not be deprecated, it was removed from sdk -> https://stackoverflow.com/a/55622713/2838206
> 
Yes I read the same thread.

> But... it also looks like Guice does not support those annotations... so you'd have to figure out a different way to manage object lifecycle.

Could you provide me some article or give me tips how to do it?

Reply:
Possible side effect would be an NPE thrown during object creation. I don't know if that's even possible, but just a thing to keep in mind.

I can't provide you a solution to that, I only read one stackoverflow thread that Guice does not "respect" those annotations. I have no real work experience with Guice, so I can't tell how it can be done.

Reply:
I choose Guice for this one to learn sth new but I see that Spring is the best :)